src/java/com/palantir/cassandra/actions/DefaultVolumeIntegrityCheckAction.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.cassandra.actions;
+
+import java.io.IOException;
+import java.util.Optional;
+import java.util.UUID;
+
+import com.palantir.cassandra.utils.FileParser;
+import com.palantir.logsafe.Preconditions;
+
+/**
+ * Check if the backing volume has been tampered with. While Cassandra supports multiple data drive, we only
+ * use 1 data drive.
+ */
+public final class DefaultVolumeIntegrityCheckAction implements Action
```

Comment:
nit

```suggestion
public final class VolumeIntegrityCheck
```

i don't see a reason to name it Default* when it's not implementing a VolumeIntegrityCheck interface and there's no way for there to be a non-default volume integrity check at the moment